Add blog content for post recap 2022 (#778)
falcosecurity/falco-website: content/en/blog/recap-2022/index.md

---
title: "Falco: Project Recap 2022"
slug: recap-2022

> date: 2022-12-20

Author: ```suggestion
date: 2023-01-04
```

> date: 2022-12-20

Reviewer: ```suggestion
date: 2023-01-03
```

author: Jacqueline Salinas
images:
  - /blog/recap-2022/images/recap-2022-featured.png
---

![](images/recap-2022-featured.png)

Dear Falco community, 

> As we write this, it’s unbelievable that 2022 is wrapping up and very soon! As always, we are truly grateful for the amazing community of maintainers and contributors. The Falco project and community would not be the vibrant ecosystem it is today without these incredible individuals. Let’s take a look at what the community accomplished in 2022, and we’ll give a quick glimpse of what to expect in 2023!  

Reviewer: We should update this section, as we're in 2023 now

Author: ```suggestion
As always, we are truly grateful for the amazing community of maintainers and contributors. The Falco project and community would not be the vibrant ecosystem it is today without these incredible individuals. Let’s take a look at what the community accomplished in 2022, and we’ll give a quick glimpse of what to expect in 2023!  
```

- [Luca Guerra](https://github.com/lucaguerra), Sysdig
- [Mark Stemm](https://github.com/mstemm), Sysdig
- [Massimiliano Giovagnoli](https://github.com/maxgio92), Clastix
- [Mauro Ezequiel Moltrasio](https://github.com/molter73), RedHat
- [Michele Zuccala](https://github.com/zuc), Sysdig
- [Radu Andries](https://github.com/admiral0), Independent
- [Teryl Taylor](https://github.com/terylt), IBM
- [Thomas Labarussias](https://github.com/issif), Sysdig

### New and noteworthy in Falco land 
The Falco community was busy in 2022, figuring out how to better serve contributors, developing new features, fixing bugs, and taking the project to new heights. Here is a recap of key milestones reached this year: 

- Falco’s governance underwent a facelift this year. Why is this important? As a CNCF project, Falco is committed to promoting a healthy community of contributors and maintainers from multiple vendors. In aid of this, the core maintainers formerly announced & published these updates to help contributors better serve the community. To read the full announcement, see [updating Falco Governance](https://falco.org/blog/updating-falco-governance/).

> 

Author: ```suggestion
At the beginning of 2022, Falco introduced a game changing feature, Falco Plugins. They allow Falco to monitor and trigger alerts for any kind of event. Since the launch of the new plugin framework the Falco community has collaborated to create plugins for GitHub, AWS CloudTrail and Okta.

A plugin also replaced the way Falco consumes the Audit Logs generated by a K8s API server. With these plugins, Falco covers more in depth the aspects of your infrastructure and allows you to use a single syntax for rules. Our adopters asked us for a way to monitor K8s Audit Logs and the previous implementation used an internal web server to receive the logs from the Kubernetes API. This method didn't support clusters managed by cloud providers, such as EKS, AKS, or GKE as they had to capture the Audit Logs for their own usage and then add them to their log aggregators. This situation is now solved thanks to the plugin framework and we're proud to have announced the first release of the plugin for EKS Audit Logs!!! Read the full blog [here!](https://falco.org/blog/k8saudit-eks-plugin/)
```

> At the beginning of 2022, Falco introduced a game changing feature, Falco Plugins. They allow Falco to monitor and trigger alerts for any kind of event. Since the launch of the new plugin framework the Falco community has collaborated to create plugins for GitHub, AWS CloudTrail and Okta.

Author: need to update this into a bullet

Author: ```suggestion
- At the beginning of 2022, Falco introduced a game changing feature, Falco Plugins. They allow Falco to monitor and trigger alerts for any kind of event. Since the launch of the new plugin framework the Falco community has collaborated to create plugins for GitHub, AWS CloudTrail and Okta.
A plugin also replaced the way Falco consumes the Audit Logs generated by a K8s API server. With these plugins, Falco covers more in depth the aspects of your infrastructure and allows you to use a single syntax for rules. Our adopters asked us for a way to monitor K8s Audit Logs and the previous implementation used an internal web server to receive the logs from the Kubernetes API. This method didn't support clusters managed by cloud providers, such as EKS, AKS, or GKE as they had to capture the Audit Logs for their own usage and then add them to their log aggregators. This situation is now solved thanks to the plugin framework and we're proud to have announced the first release of the plugin for EKS Audit Logs!!! Read the full blog [here!](https://falco.org/blog/k8saudit-eks-plugin/)
```

> A plugin also replaced the way Falco consumes the Audit Logs generated by a K8s API server. With these plugins, Falco covers more in depth the aspects of your infrastructure and allows you to use a single syntax for rules. Our adopters asked us for a way to monitor K8s Audit Logs and the previous implementation used an internal web server to receive the logs from the Kubernetes API. This method didn't support clusters managed by cloud providers, such as EKS, AKS, or GKE as they had to capture the Audit Logs for their own usage and then add them to their log aggregators. This situation is now solved thanks to the plugin framework and we're proud to have announced the first release of the plugin for EKS Audit Logs!!! Read the full blog [here!](https://falco.org/blog/k8saudit-eks-plugin/)

Author: the format is messed up and needs to be indented
this should also be part of the bullet from line 47

Author: ```suggestion

```

- Giant Swarm simplifies the maintenance of the software stack within Kubernetes clusters by using its App Platform technology. Leverage this to easily deploy Falco, either individually or as part of Giant Swarm's Security Pack, to secure a managed Kubernetes service. [Learn how to deploy and manage applications across hybrid environments.](https://falco.org/blog/giantswarm-app-platform-falco/)

> - At the beginning of 2022, Falco introduced a game changing feature, Falco Plugins. They allow Falco to monitor and trigger alerts for any kind of event. Since the launch of the new plugin framework the Falco community has collaborated to create plugins for GitHub, AWS CloudTrail and Okta. 

Author: ```suggestion
```

Author: removing and moving up higher on the list (line 46 - 48)

> - A plugin also replaced the way Falco consumes the Audit Logs generated by a K8s API server. With these plugins, Falco covers more in depth the aspects of your infrastructure and allows you to use a single syntax for rules. Our adopters asked us for a way to monitor K8s Audit Logs and the previous implementation used an internal web server to receive the logs from the Kubernetes API. This method didn't support clusters managed by cloud providers, such as EKS, AKS, or GKE as they had to capture the Audit Logs for their own usage and then add them to their log aggregators. This situation is now solved thanks to the plugin framework and we're proud to have announced the first release of the plugin for EKS Audit Logs!!! Read the full blog [here! ](https://falco.org/blog/k8saudit-eks-plugin/) 

Author: ```suggestion
```

Author: see line 48

  - Introduced the Kernel Crawler, a new tool that automatically identifies the most up to date kernel versions supported by popular distros
  - Syscall kernel ring-buffer size is now customizable for your environment needs
  - Mitigations for libsinsp’s Kubernetes metadata client to address recent issues that caused Falco to crash
  - Support for multiple simultaneous event sources, which means that you can now run multiple event sources in the same Falco instance
  - Added minikube as a supported platform in the driver loader and included it in our driver build matrix
  - Rule alert rate limiter is now optional and disabled at default
  - Support for two new syscalls and many improvements to the default Falco security ruleset

### Falcosidekick & UI
[Falcosidekick](https://github.com/falcosecurity/falcosidekick) is a little daemon that extends a number of possible outputs, and since its creation this little guy has evolved in many amazing ways. A big thank you to Thomas Labarussias for driving the development of Falcosidekick and the UI. Here is how the project evolved in 2022: 

> - Falcosidekick saw a major update in June of 2022, delivering four new outputs and enhancing existing ones too! Not only did we see updates in outputs, but the community delivered the first version of the Falcosidekick UI. The UI offered these new features: Redis DB for long term storage of events, API for counting and searching for events, and lastly filters to keep and share query strings. For a full detailed account of this update, see the blog Falcosidekick 2.25.0 and Falcosidekick 2.0.0. 

Author: For a full detailed account of this update, see the blog [Falcosidekick 2.25.0 and Falcosidekick 2.0.0. ](https://falco.org/blog/falcosidekick-2-25-0-falco-2-0-0/)

Author: ```suggestion
- Falcosidekick saw a major update in June of 2022, delivering four new outputs and enhancing existing ones too! Not only did we see updates in outputs, but the community delivered the first version of the Falcosidekick UI. The UI offered these new features: Redis DB for long term storage of events, API for counting and searching for events, and lastly filters to keep and share query strings. For a full detailed account of this update, see the blog [Falcosidekick 2.25.0 and Falcosidekick 2.0.0.](https://falco.org/blog/falcosidekick-2-25-0-falco-2-0-0/).
```

> - Falcosidekick saw a major update in June of 2022, delivering four new outputs and enhancing existing ones too! Not only did we see updates in outputs, but the community delivered the first version of the Falcosidekick UI. The UI offered these new features: Redis DB for long term storage of events, API for counting and searching for events, and lastly filters to keep and share query strings. For a full detailed account of this update, see the blog Falcosidekick 2.25.0 and Falcosidekick 2.0.0. 

Reviewer: ```suggestion
- Falcosidekick saw a major update in June of 2022, delivering four new outputs and enhancing existing ones too! Not only did we see updates in outputs, but the community delivered the first version of the Falcosidekick UI. The UI offered these new features: Redis DB for long term storage of events, API for counting and searching for events, and lastly filters to keep and share query strings. For a full detailed account of this update, see the blog [Falcosidekick 2.25.0 and Falcosidekick 2.0.0](https://falco.org/blog/falcosidekick-2-25-0-falco-2-0-0/). 
```

- The field extraction capability, which parse the data coming from source plugins and expose new fields that can be used in Falco rules.

By supporting one or both capabilities, plugins allows users to feed data into Falco, parse it in useful ways and create rules and policies from it, read the full announcement [here!](https://falco.org/blog/falco-announcing-plugins/) 

Official plugins launched in 2022 include: 
- AWS CloudTrail
- GitHub
- K8saudit
- K8saudit-EKS
- Okta

### Falco project growth
The Falco project excitedly announced that on Nov 4, 2022, project maintainers submitted the proposal to be considered for graduation. This milestone is important because it demonstrates the health and maturity of the project to the Kubernetes & cloud ecosystem.

Falco joined the CNCF as a Sandbox project in the fall of 2018 and evolved into Incubating status two years later. The community believes the next natural step is for Falco to apply for graduation. The project has reached maturity in community health, diversity of contributions & contributors, and ecosystem adoption.

- Project highlights for 2022

> - Github stars: + 26%

Author: ```suggestion
- Github stars: + 31%
```

- Core repo contributors: +65%
- Forks: +7%
- Slack channel members: +126%
- Downloads:
  - 41.7M Docker hub pulls (25% growth)
  - 1.2M AWS ECR

### New features in Falco
The most significant additions since acceptance for Incubation include:

- Stable release schedule
- Contribution of the whole libs source code into the Falcosecurity organization
- Plugin framework, allowing event sources other than syscalls
- New eBPF probe
- gVisor integration

Thank you to all of the contributors and maintainers of the Falco project. The project would not be where it is today without the help and dedication of these key individuals. 

> See you in 2023, 

Author: ```suggestion
Wishing you a happy 2023, 
```

- [Luca Guerra](https://github.com/lucaguerra), Sysdig
- [Mark Stemm](https://github.com/mstemm), Sysdig
- [Massimiliano Giovagnoli](https://github.com/maxgio92), Clastix
- [Mauro Ezequiel Moltrasio](https://github.com/molter73), RedHat
- [Michele Zuccala](https://github.com/zuc), Sysdig
- [Radu Andries](https://github.com/admiral0), Independent
- [Teryl Taylor](https://github.com/terylt), IBM
- [Thomas Labarussias](https://github.com/issif), Sysdig

### New and noteworthy in Falco land 
The Falco community was busy in 2022, figuring out how to better serve contributors, developing new features, fixing bugs, and taking the project to new heights. Here is a recap of key milestones reached this year: 

- Falco’s governance underwent a facelift this year. Why is this important? As a CNCF project, Falco is committed to promoting a healthy community of contributors and maintainers from multiple vendors. In aid of this, the core maintainers formerly announced & published these updates to help contributors better serve the community. To read the full announcement, see [updating Falco Governance](https://falco.org/blog/updating-falco-governance/).

> 

Author: - At the beginning of 2022, Falco introduced a game changing feature, Falco Plugins. They allow Falco to monitor and trigger alerts for any kind of event. Since the launch of the new plugin framework the Falco community has collaborated to create plugins for GitHub, AWS CloudTrail and Okta.

> 

Author: - A plugin also replaced the way Falco consumes the Audit Logs generated by a K8s API server. With these plugins, Falco covers more in depth the aspects of your infrastructure and allows you to use a single syntax for rules. Our adopters asked us for a way to monitor K8s Audit Logs and the previous implementation used an internal web server to receive the logs from the Kubernetes API. This method didn't support clusters managed by cloud providers, such as EKS, AKS, or GKE as they had to capture the Audit Logs for their own usage and then add them to their log aggregators. This situation is now solved thanks to the plugin framework and we're proud to have announced the first release of the plugin for EKS Audit Logs!!! Read the full blog [here! ](https://falco.org/blog/k8saudit-eks-plugin/)

> - Feeling terrorized by sneaky cryptominer programs? Falco is helping fight these monsters. Cryptominers are programs that utilize computer resources to mine cryptocurrency. XMRig is an example of an open source crypto mining software designed for the sole purpose of mining cryptocurrencies, like Monero or Bitcoin. Cryptominers usually get rewarded with a token for every successful transaction mined, which makes cryptomining a profitable activity.

Author: it demonstrates community health and growth - let's keep it.

- As the home of such important assets, GitHub repositories should be at the top of your list of security priorities. However, many people fail to put in place even basic measures to protect source code repositories. [Read abou](https://falco.org/blog/falco-plugin-github/)t three risks to GitHub repos and how you can reliably detect them, as they happen, using Falco open source security. 

- Before 0.32.1, Falco could not work with gVisor monitored sandboxes because it was not possible to install a kernel module or eBPF probe in such an environment. But wouldn't it be great to leverage the stream of system call information that gVisor collects through its powerful monitoring system directly in Falco? This is exactly what became possible with gVisor release 20220704.0 and Falco 0.32.1. [Learn how to integrate gVisor and Falco on Docker](https://falco.org/blog/intro-gvisor-falco/). 

> - With Falco 0.32.1, you can now run Falco on Apple ARM M1 CPUs. It requires a Linux virtual machine (VM) since Falco doesn't run on OSX, but it is pretty straightforward. Here are the step by step [instructions.](https://falco.org/blog/falco-apple-silicon/) 

Reviewer 2: I think the big news is support for ARM in general, I'd eventually mention the Apple Silicon compatibility as a +1 example.

- Giant Swarm simplifies the maintenance of the software stack within Kubernetes clusters by using its App Platform technology. Leverage this to easily deploy Falco, either individually or as part of Giant Swarm's Security Pack, to secure a managed Kubernetes service. [Learn how to deploy and manage applications across hybrid environments.](https://falco.org/blog/giantswarm-app-platform-falco/)

> - At the beginning of 2022, Falco introduced a game changing feature, Falco Plugins. They allow Falco to monitor and trigger alerts for any kind of event. Since the launch of the new plugin framework the Falco community has collaborated to create plugins for GitHub, AWS CloudTrail and Okta. 

Reviewer 2: From a logical perspective, I would mention the plugin system before the GitHub security bullet point above, since that is implemented through the GitHub plugin.

Author: updated!

> - At the beginning of 2022, Falco introduced a game changing feature, Falco Plugins. They allow Falco to monitor and trigger alerts for any kind of event. Since the launch of the new plugin framework the Falco community has collaborated to create plugins for GitHub, AWS CloudTrail and Okta. 

Author: i've added it to line 46 and we can delete this portion

> - A plugin also replaced the way Falco consumes the Audit Logs generated by a K8s API server. With these plugins, Falco covers more in depth the aspects of your infrastructure and allows you to use a single syntax for rules. Our adopters asked us for a way to monitor K8s Audit Logs and the previous implementation used an internal web server to receive the logs from the Kubernetes API. This method didn't support clusters managed by cloud providers, such as EKS, AKS, or GKE as they had to capture the Audit Logs for their own usage and then add them to their log aggregators. This situation is now solved thanks to the plugin framework and we're proud to have announced the first release of the plugin for EKS Audit Logs!!! Read the full blog [here! ](https://falco.org/blog/k8saudit-eks-plugin/) 

Author: we can remove this section and move up to point 1 or 2

### Core releases 

> The Falco [release process has been well-defined](https://github.com/falcosecurity/falco/blob/master/RELEASE.md) with a fixed schedule of three releases per year (roughly in Jan, May, and Sept), plus patch releases that can occur when needed upon agreement by maintainers (for example, for hotfixes or security patches). Here are the three major releases for 2022: 

Reviewer 2: ```suggestion
The Falco [release process has been well-defined](https://github.com/falcosecurity/falco/blob/master/RELEASE.md) with a fixed schedule of three releases per year (roughly at the end of Jan, May, and Sept), plus patch releases that can occur when needed upon agreement by maintainers (for example, for hotfixes or security patches). Here are the three major releases for 2022: 
```

  - Added minikube as a supported platform in the driver loader and included it in our driver build matrix
  - Rule alert rate limiter is now optional and disabled at default
  - Support for two new syscalls and many improvements to the default Falco security ruleset

### Falcosidekick & UI
[Falcosidekick](https://github.com/falcosecurity/falcosidekick) is a little daemon that extends a number of possible outputs, and since its creation this little guy has evolved in many amazing ways. A big thank you to Thomas Labarussias for driving the development of Falcosidekick and the UI. Here is how the project evolved in 2022: 

- Falcosidekick saw a major update in June of 2022, delivering four new outputs and enhancing existing ones too! Not only did we see updates in outputs, but the community delivered the first version of the Falcosidekick UI. The UI offered these new features: Redis DB for long term storage of events, API for counting and searching for events, and lastly filters to keep and share query strings. For a full detailed account of this update, see the blog Falcosidekick 2.25.0 and Falcosidekick 2.0.0. 

> - Well, we thought the June release was big, but Thomas had some goodies for the community just in time for Christmas! The number of new outputs for this release include: Yandex, MQTT & Node-Red, plus many more totaling to eight new outputs! However, it’s not just new outputs that were added but also enhancements to the existing outputs were included in this release. The new features for Falcosidekick 2.27.0 also include: support for both SMTP & Kafka, which can now benefit from SASL auth mechanisms, now you can use environment variables to customize your fields, create policy reports, among a few other very cool features. To read the full release notes, take a look at  [Falcosidekick 2.27.0 and Falcosidekick-UI 2.1.0](https://falco.org/blog/falcosidekick-2-27-0-ui-2-1-0/).

Author: ```suggestion
```
Remove all of this section.

### Plugins 
Falco v0.31.0 resulted in many exciting new features. One that is particularly strategic for the project is the general availability of the plugins framework. Why are plugins exciting and what do they mean for the future of Falco? 

> Plugins are shared libraries that can be loaded by Falco to extend its functionality. Plugins come in two flavors:
> 
> - Source Plugins, which add new data sources to Falco. They produce input events, from either the local machine or a remote source, that Falco can understand.
> - Extractor Plugins, which parse the data coming from source plugins and expose new fields that can be used in Falco rules.
> 
> The combination of Source and Extractor plugins allows users to feed data into Falco, parse it in useful ways and create rules and policies from it, read the full announcement [here!](https://falco.org/blog/falco-announcing-plugins/) 

Reviewer 2: ```suggestion
Plugins are shared libraries that can be loaded by Falco to extend its functionality. Plugins can currently implement two capabilities:

- The event sourcing capability, which add new data sources to Falco. They produce input events, from either the local machine or a remote source, that Falco can understand.
- The field extraction capability, which parse the data coming from source plugins and expose new fields that can be used in Falco rules.

By supporting one or both capabilities, plugins allows users to feed data into Falco, parse it in useful ways and create rules and policies from it, read the full announcement [here!](https://falco.org/blog/falco-announcing-plugins/) 
```

Official plugins launched in 2022 include: 
- AWS CloudTrail
- GitHub
- K8saudit
- K8saudit-EKS
- Okta

### Falco project growth
The Falco project excitedly announced that on Nov 4, 2022, project maintainers submitted the proposal to be considered for graduation. This milestone is important because it demonstrates the health and maturity of the project to the Kubernetes & cloud ecosystem.

Falco joined the CNCF as a Sandbox project in the fall of 2018 and evolved into Incubating status two years later. The community believes the next natural step is for Falco to apply for graduation. The project has reached maturity in community health, diversity of contributions & contributors, and ecosystem adoption.

> - Project highlights for 2022

Reviewer 2: How are these calculated? For example, looking at [Falco's devstats](https://falco.devstats.cncf.io/d/3/stars-and-forks-by-repository?orgId=1&from=1638313200000&to=1669849200000) the growth in stars and forks across all repos are ~+31% and ~+46% since last year.

Author: took these from the dd doc

> - Github stars: + 26%

Author: - Github stars: ~+31%

- Core repo contributors: +65%

> - Forks: +7%

Author: - Forks: ~+46%

Author: ```suggestion
- Forks: +46%
```
